components/jsx/preferences-modal/index.js

```
@@ -29,17 +40,38 @@ const positionModal = ({ eventTrigger, preferencesModal }) => {
 };
 
 const preferenceModalShowAndHide = ({ event, preferencesModal }) => {
-	if (!event.target) {
-		return;
-	}
-
 	preferencesModal.classList.toggle('n-myft-ui__preferences-modal--show');
 
 	if (preferencesModal.classList.contains('n-myft-ui__preferences-modal--show')) {
-		positionModal({ eventTrigger: event.target, preferencesModal });
+		positionModal({ event, preferencesModal });
+	} else {
+		// Remove existing errors when hiding the modal
+		renderError('');
 	}
 };
 
+const renderError = ({ message, preferencesModal }) => {
+	const errorElement = preferencesModal.querySelector('[data-component-id="myft-preference-modal-error"]');
```

Non blocking thought: maybe we can return early if `errorElement` has not been found

---

Yeah I am actually trying to think if this is the right thing to do or not (along with some other places where I have exited early)

It might be helpful to let it error/throw in some cases so we can better identify the errors instead of them becoming silent/unnoticed

---

So lets discuss this later and come up with an approach

---

Agreed. But where would we log these errors?

---

Currently client side errors still go to Sentry (I think)